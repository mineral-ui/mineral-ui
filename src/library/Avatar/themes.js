/* @flow */
import type { AvatarThemeFn } from './types';

export const avatarTheme: AvatarThemeFn = (baseTheme) => ({
  Avatar_fontSize_small: baseTheme.fontSize_mouse,
  Avatar_fontSize_medium: baseTheme.fontSize_ui,
  Avatar_fontSize_large: baseTheme.h4_fontSize,
  Avatar_fontSize_jumbo: baseTheme.h4_fontSize,
  Avatar_fontWeight: baseTheme.fontWeight_bold,
  Avatar_size_small: baseTheme.size_small,
  Avatar_size_medium: baseTheme.size_medium,
  Avatar_size_large: baseTheme.size_large,
  Avatar_size_jumbo: baseTheme.size_jumbo,

  ...baseTheme
});
