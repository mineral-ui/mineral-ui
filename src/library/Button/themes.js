/* @flow */
import { pxToEm } from '../styles';

import type { ButtonThemeFn } from './types';

// prettier-ignore
export const buttonTheme: ButtonThemeFn = (baseTheme) => ({
  Button_backgroundColor: baseTheme.backgroundColor,
  Button_backgroundColor_active: baseTheme.backgroundColor_active,
  Button_backgroundColor_focus: baseTheme.backgroundColor_focus,
  Button_backgroundColor_hover: baseTheme.backgroundColor_hover,
  Button_backgroundColor_minimal_active: baseTheme.backgroundColor_active,
  Button_backgroundColor_minimal_hover: baseTheme.backgroundColor_hover,
  Button_backgroundColor_primary: baseTheme.backgroundColor_themePrimary,
  Button_backgroundColor_primary_active: baseTheme.backgroundColor_themePrimary_active,
  Button_backgroundColor_primary_focus: baseTheme.backgroundColor_themePrimary_focus,
  Button_backgroundColor_primary_hover: baseTheme.backgroundColor_themePrimary_hover,
  Button_borderColor: baseTheme.borderColor,
  Button_borderColor_active: baseTheme.borderColor_theme_active,
  Button_borderColor_focus: baseTheme.borderColor_theme_focus,
  Button_borderColor_hover: baseTheme.borderColor_theme_hover,
  Button_borderRadius: baseTheme.borderRadius_3,
  Button_borderWidth: 1, // px,
  Button_boxShadow_focus: `0 0 0 1px ${baseTheme.boxShadow_focusInner}, 0 0 0 2px ${baseTheme.borderColor_theme_focus}`,
  Button_color: baseTheme.color_theme,
  Button_color_minimal: baseTheme.color_theme,
  Button_color_primary: baseTheme.color_themePrimary,
  Button_fontWeight: baseTheme.fontWeight_semiBold,
  Button_paddingHorizontal: baseTheme.space_inset_sm_plus,
  Button_paddingIconOnly_small: pxToEm(3),
  Button_paddingIconOnly_medium: pxToEm(7),
  Button_paddingIconOnly_large: pxToEm(7),
  Button_paddingIconOnly_jumbo: pxToEm(13),
  Button_height_small: baseTheme.size_small,
  Button_height_medium: baseTheme.button_size_medium,
  Button_height_large: baseTheme.button_size_large,
  Button_height_jumbo: baseTheme.button_size_jumbo,

  ButtonContent_fontSize: baseTheme.fontSize_ui_sm,
  ButtonContent_fontSize_small: pxToEm(12),

  ButtonIcon_color: baseTheme.icon_color_theme,
  ButtonIcon_margin: baseTheme.space_inset_sm,

  ...baseTheme
});
