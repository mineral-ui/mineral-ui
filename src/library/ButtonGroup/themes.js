/* @flow */
import type { ButtonGroupThemeFn } from './types';

export const buttonGroupTheme: ButtonGroupThemeFn = (baseTheme) => ({
  ButtonGroupButton_backgroundColor_checkedDisabled: baseTheme.color_gray_40,
  ButtonGroupButton_border_disabled: `solid 1px ${baseTheme.borderColor}`,
  ButtonGroupButton_borderColor_active: baseTheme.borderColor_theme_active,
  ButtonGroupButton_borderColor_hover: baseTheme.borderColor_theme_hover,
  ButtonGroupButton_borderStartColor: baseTheme.borderColor,
  ButtonGroupButton_borderStartColor_checked: 'currentcolor',
  ButtonGroupButton_color_checkedDisabled: baseTheme.color_gray_60,

  ...baseTheme
});
