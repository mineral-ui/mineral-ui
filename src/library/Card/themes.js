/* @flow */
import { pxToEm } from '../styles';

import type {
  CardThemeFn,
  CardActionsThemeFn,
  CardBlockThemeFn,
  CardDividerThemeFn,
  CardFooterThemeFn,
  CardStatusThemeFn,
  CardTitleThemeFn
} from './types';

// prettier-ignore
export const cardTheme: CardThemeFn = (baseTheme) => ({
  Card_backgroundColor: baseTheme.panel_backgroundColor,
  Card_borderColor: baseTheme.panel_borderColor,
  Card_borderRadius: baseTheme.borderRadius_1,
  Card_boxShadow: baseTheme.boxShadow_2,
  Card_boxShadow_focus: `0 0 0 2px ${baseTheme.borderColor_theme_focus}, ${baseTheme.boxShadow_1}`,

  CardRow_marginVertical: baseTheme.space_stack_md,
  CardRow_marginVerticalLast: baseTheme.space_stack_lg,
  CardRow_paddingHorizontal: baseTheme.space_inset_md,

  ...baseTheme
});

export const cardActionsTheme: CardActionsThemeFn = (baseTheme) => ({
  CardActionsAction_spaceInline: baseTheme.space_inline_sm,

  ...baseTheme
});

export const cardBlockTheme: CardBlockThemeFn = (baseTheme) => ({
  CardBlock_fontSize: baseTheme.fontSize_ui,
  CardBlock_lineHeight: baseTheme.lineHeight_prose,

  ...baseTheme
});

export const cardDividerTheme: CardDividerThemeFn = (baseTheme) => ({
  CardDivider_borderColor: baseTheme.borderColor,
  CardDivider_borderWidth: '1px',

  ...baseTheme
});

export const cardFooterTheme: CardFooterThemeFn = (baseTheme) => ({
  CardFooter_backgroundColor: baseTheme.well_backgroundColor,
  CardFooter_borderColor: baseTheme.borderColor,

  CardFooterRow_marginVertical: baseTheme.space_stack_sm,
  CardFooterRow_marginVerticalLast: baseTheme.space_stack_md,

  CardFooterTitle_color: baseTheme.h5_color,
  CardFooterTitle_fontSize: baseTheme.h5_fontSize,
  CardFooterTitle_fontWeight: baseTheme.h5_fontWeight,

  ...baseTheme
});

export const cardStatusTheme: CardStatusThemeFn = (baseTheme) => ({
  CardStatus_fontSize: baseTheme.fontSize_ui,
  CardStatus_fontWeight: baseTheme.fontWeight_regular,

  CardStatusIcon_margin: baseTheme.space_inline_sm,
  CardStatusIcon_size: pxToEm(12),

  ...baseTheme
});

export const cardTitleTheme: CardTitleThemeFn = (baseTheme) => ({
  CardTitle_color: baseTheme.h4_color,
  CardTitle_fontSize: baseTheme.h4_fontSize,
  CardTitle_fontWeight: baseTheme.h4_fontWeight,

  CardTitleAvatar_margin: baseTheme.space_inline_sm,
  CardTitleAvatarSize: baseTheme.size_small,
  CardTitleAvatarSize_large: baseTheme.size_large,

  CardTitleIcon_margin: baseTheme.space_inline_sm,

  CardTitleSecondaryText_color: baseTheme.color_mouse,
  CardTitleSecondaryText_fontSize: baseTheme.fontSize_mouse,
  CardTitleSecondaryText_fontWeight: baseTheme.fontWeight_regular,

  CardSubtitle_color: baseTheme.color_mouse,
  CardSubtitle_fontSize: baseTheme.fontSize_mouse,
  CardSubtitle_fontWeight: baseTheme.fontWeight_regular,
  CardSubtitle_marginTop: baseTheme.space_stack_sm,

  ...baseTheme
});
