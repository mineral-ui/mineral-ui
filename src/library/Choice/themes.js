/* @flow */
import { pxToEm } from '../styles';

import type { ChoiceThemeFn, ChoiceGroupThemeFn } from './types';

export const choiceTheme: ChoiceThemeFn = (baseTheme) => {
  const colors = {
    background: baseTheme.input_backgroundColor,
    regular: baseTheme.borderColor,
    checked: baseTheme.borderColor_theme,
    focus: baseTheme.borderColor_theme_focus,
    hover: baseTheme.borderColor_theme_hover
  };

  // prettier-ignore
  return {
    ChoiceControl_backgroundColor: colors.background,
    ChoiceControl_backgroundColor_checked: colors.checked,
    ChoiceControl_backgroundColor_checkedHover: colors.hover,
    ChoiceControl_borderColor: colors.regular,
    ChoiceControl_borderColor_hover: colors.hover,
    ChoiceControl_borderColor_checked: colors.checked,
    ChoiceControl_borderColor_checkedHover: colors.hover,
    ChoiceControl_borderRadius: baseTheme.borderRadius_1,
    ChoiceControl_boxShadow_focus: `0 0 0 1px ${baseTheme.boxShadow_focusInner}, 0 0 0 2px ${colors.focus}`,
    ChoiceControl_size: pxToEm(16),
    ChoiceControl_size_jumbo: pxToEm(24),

    ChoiceText_color: baseTheme.color,
    ChoiceText_fontSize: baseTheme.fontSize_ui,
    ChoiceText_fontSize_small: pxToEm(12),
    ChoiceText_marginHorizontal: baseTheme.space_inline_md,

    ...baseTheme
  };
};

export const choiceGroupTheme: ChoiceGroupThemeFn = (baseTheme) => ({
  ChoiceGroupControl_marginHorizontal_inline: baseTheme.space_inline_xl,
  ChoiceGroupControl_marginVertical_stacked: baseTheme.space_stack_md,
  ChoiceGroupControl_marginVertical_stackedJumbo: baseTheme.space_stack_lg,

  ...baseTheme
});
