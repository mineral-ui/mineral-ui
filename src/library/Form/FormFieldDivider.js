/* @flow */
import React from 'react';
import { FormFieldDividerRoot as Root } from './styled';

import { formFieldDividerPropTypes } from './propTypes';
import type { FormFieldDividerProps } from './types';

const FormFieldDivider = (props: FormFieldDividerProps) => (
  <Root {...props} role="separator" />
);

FormFieldDivider.propTypes = formFieldDividerPropTypes;

export default FormFieldDivider;
