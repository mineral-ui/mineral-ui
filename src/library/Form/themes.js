/* @flow */

import type {
  FormFieldThemeFn,
  FormFieldDividerThemeFn,
  FormFieldsetThemeFn
} from './types';

// [1] This deviates from h6 because it must look right when used both within
//     and without a FormFieldset
export const formFieldTheme: FormFieldThemeFn = (baseTheme) => ({
  FormFieldCaption_color: baseTheme.color_mouse,
  FormFieldCaption_fontSize: baseTheme.fontSize_mouse,
  FormFieldCaption_marginTop: baseTheme.space_stack_xxs,
  FormFieldCaption_marginTop_isGroup: baseTheme.space_stack_sm,

  FormFieldLabel_color: baseTheme.h5_color,
  FormFieldLabel_fontSize: baseTheme.h6_fontSize,
  FormFieldLabel_fontWeight: baseTheme.fontWeight_semiBold, // [1]
  FormFieldLabel_marginBottom: baseTheme.space_stack_sm,

  FormFieldSecondaryText_fontSize: baseTheme.fontSize_mouse,
  FormFieldSecondaryText_color: baseTheme.color_mouse,
  FormFieldSecondaryText_color_required: baseTheme.color_required,
  FormFieldSecondaryText_fontWeight: baseTheme.fontWeight_regular,

  ...baseTheme
});

export const formFieldDividerTheme: FormFieldDividerThemeFn = (baseTheme) => ({
  FormFieldDivider_borderColor: baseTheme.borderColor,
  FormFieldDivider_borderWidth: '1px',
  FormFieldDivider_margin: baseTheme.space_stack_sm,

  ...baseTheme
});

export const formFieldsetTheme: FormFieldsetThemeFn = (baseTheme) => ({
  FormFieldset_borderColor: baseTheme.borderColor_transparent,

  FormFieldsetLegend_color: baseTheme.h5_color,
  FormFieldsetLegend_fontSize: baseTheme.h5_fontSize,
  FormFieldsetLegend_fontWeight: baseTheme.h5_fontWeight,

  ...baseTheme
});
