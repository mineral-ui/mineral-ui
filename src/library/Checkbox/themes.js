/* @flow */
import { mapComponentThemes } from '../themes';
import { choiceTheme, choiceGroupTheme } from '../Choice';

import type { CheckboxThemeFn, CheckboxGroupThemeFn } from './types';

import { pxToEm } from '../styles';

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const checkboxTheme: CheckboxThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'Choice',
      theme: choiceTheme(baseTheme)
    },
    {
      name: 'Checkbox',
      theme: {
        CheckboxControl_backgroundColor_checked: baseTheme.checkbox_backgroundColor_checked,
        CheckboxControl_backgroundColor_checkedHover: baseTheme.checkbox_backgroundColor_checkedHover,
        CheckboxControl_borderColor: baseTheme.checkbox_borderColor,
        CheckboxControl_borderColor_checked: baseTheme.checkbox_borderColor_checked,
        CheckboxControl_borderColor_checkedHover: baseTheme.checkbox_borderColor_checkedHover,
        CheckboxControl_borderColor_hover: baseTheme.checkbox_borderColor_hover,
        CheckboxControl_borderRadius: baseTheme.borderRadius_2,
        CheckboxControl_borderWidth: `${baseTheme.checkbox_borderWidth}px`,
        CheckboxControl_size: pxToEm(20),
        CheckboxText_marginHorizontal: baseTheme.space_inline_sm
      }
    },
    baseTheme
  );

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const checkboxGroupTheme: CheckboxGroupThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'ChoiceGroup',
      theme: choiceGroupTheme(baseTheme)
    },
    {
      name: 'CheckboxGroup',
      theme: {}
    },
    baseTheme
  );
