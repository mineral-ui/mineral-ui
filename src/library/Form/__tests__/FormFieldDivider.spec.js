/* @flow */
import React from 'react';
import { shallow } from 'enzyme';
import { formFieldDividerTheme } from '../themes';
import FormFieldDivider from '../FormFieldDivider';
import examples from '../../../website/app/demos/Form/FormFieldDivider/examples';
import testDemoExamples from '../../../../utils/testDemoExamples';
import testThemeOverrides from '../../../../utils/testThemeOverrides';
import { getProcessedComponentThemeKeys } from '../../themes/processComponentTheme';

function shallowFormFieldDivider(props = {}) {
  const formFieldDividerProps = {
    ...props
  };

  return shallow(<FormFieldDivider {...formFieldDividerProps} />);
}

describe('FormFieldDivider', () => {
  testDemoExamples(examples);

  describe('renders', () => {
    it('root', () => {
      const formFieldDivider = shallowFormFieldDivider();

      expect(formFieldDivider.exists()).toEqual(true);
    });
  });

  describe('theme overrides', () => {
    testThemeOverrides(
      <FormFieldDivider />,
      getProcessedComponentThemeKeys(formFieldDividerTheme)
    );
  });
});
