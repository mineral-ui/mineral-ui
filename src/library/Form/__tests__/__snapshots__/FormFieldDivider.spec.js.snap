// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFieldDivider demo examples basic 1`] = `
.css-0,
[data-css-0] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
}

.css-0 *,
[data-css-0] *,
.css-0 *::before,
[data-css-0] *::before,
.css-0 *::after,
[data-css-0] *::after {
  box-sizing: inherit;
}

.css-1,
[data-css-1] {
  color: #58606e;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  font-size: 0.875em;
  font-weight: 600;
  justify-content: space-between;
  margin-bottom: 0.5em;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
}

.css-1 > *,
[data-css-1] > * {
  align-self: flex-end;
  display: inline-block;
  -webkit-align-self: flex-end;
}

.css-2,
[data-css-2] {
  align-items: center;
  cursor: text;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  width: 100%;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-2 [role="img"],
[data-css-2] [role="img"] {
  display: block;
  fill: #c8d1e0;
  flex: 0 0 auto;
  margin: 0 0.5em;
  -webkit-flex: 0 0 auto;
}

.css-2 [role="img"]:last-of-type,
[data-css-2] [role="img"]:last-of-type {
  fill: #c8d1e0;
}

.css-4,
[data-css-4] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  position: relative;
  z-index: 1;
  align-items: center;
  cursor: text;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  width: 100%;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-4 *,
[data-css-4] *,
.css-4 *::before,
[data-css-4] *::before,
.css-4 *::after,
[data-css-4] *::after {
  box-sizing: inherit;
}

.css-4:hover > div:last-child,
[data-css-4]:hover > div:last-child,
.css-4[data-simulate-hover] > div:last-child,
[data-css-4][data-simulate-hover] > div:last-child {
  border-color: #2e6fd9;
}

.css-4:focus > div:last-child,
[data-css-4]:focus > div:last-child,
.css-4[data-simulate-focus] > div:last-child,
[data-css-4][data-simulate-focus] > div:last-child {
  border-color: #c8d1e0;
  box-shadow: 0 0 0 1px #fff, 0 0 0 2px #2e6fd9;
}

.css-4:active > div:last-child,
[data-css-4]:active > div:last-child,
.css-4[data-simulate-active] > div:last-child,
[data-css-4][data-simulate-active] > div:last-child {
  border-color: #c8d1e0;
  box-shadow: 0 0 0 1px #fff, 0 0 0 2px #2e6fd9;
}

.css-4 [role="img"],
[data-css-4] [role="img"] {
  display: block;
  fill: #c8d1e0;
  flex: 0 0 auto;
  margin: 0 0.5em;
  -webkit-flex: 0 0 auto;
}

.css-4 [role="img"]:last-of-type,
[data-css-4] [role="img"]:last-of-type {
  fill: #c8d1e0;
}

.css-15,
[data-css-15] {
  background-color: transparent;
  border: 0;
  box-shadow: none;
  flex: 1 1 auto;
  font-family: inherit;
  min-width: 0;
  width: 100%;
  color: #58606e;
  font-size: 0.875em;
  height: 2.857142857142857em;
  outline: 0;
  padding-left: 1.1428571428571428em;
  padding-right: 1.1428571428571428em;
  -webkit-flex: 1 1 auto;
}

.css-15::placeholder,
[data-css-15]::placeholder,
.css-15[data-simulate-placeholder],
[data-css-15][data-simulate-placeholder] {
  color: #8e99ab;
  font-style: italic;
}

.css-15::-ms-input-placeholder,
[data-css-15]::-ms-input-placeholder,
.css-15[data-simulate-msinputplaceholder],
[data-css-15][data-simulate-msinputplaceholder] {
  color: #8e99ab;
  font-style: italic;
}

.css-15:-ms-input-placeholder,
[data-css-15]:-ms-input-placeholder,
.css-15[data-simulate-msinputplaceholder],
[data-css-15][data-simulate-msinputplaceholder] {
  color: #8e99ab;
  font-style: italic;
}

.css-15::-ms-clear,
[data-css-15]::-ms-clear,
.css-15[data-simulate-msclear],
[data-css-15][data-simulate-msclear] {
  display: none;
}

.css-15:focus ~ div:last-child,
[data-css-15]:focus ~ div:last-child,
.css-15[data-simulate-focus] ~ div:last-child,
[data-css-15][data-simulate-focus] ~ div:last-child {
  border-color: #c8d1e0;
  box-shadow: 0 0 0 1px #fff, 0 0 0 2px #2e6fd9;
}

.css-29,
[data-css-29] {
  background-color: #fff;
  border-color: #c8d1e0;
  border-radius: 0.1875em;
  border-style: solid;
  border-width: 1px;
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  z-index: -1;
}

.css-30,
[data-css-30] {
  background-color: #c8d1e0;
  height: 1px;
  margin: 0.5em 0;
}

<glamorous(div)>
  <div
    className="css-1p1xtzc"
  >
    <FormField
      label="Old Password"
      requiredText="Required"
    >
      <FormField>
        <div
          className="css-0"
        >
          <label>
            <glamorous(div)
              key="formField-1-textWrapper"
            >
              <div
                className="css-1"
              >
                <span
                  id="formField-1-labelText"
                >
                  Old Password
                </span>
              </div>
            </glamorous(div)>
            <TextInput
              key="formField-1-control"
              type="password"
            >
              <TextInput
                control={[Function]}
                controlProps={
                  Object {
                    "aria-describedby": undefined,
                    "aria-invalid": undefined,
                    "aria-required": undefined,
                    "controlRef": undefined,
                    "disabled": undefined,
                    "readOnly": undefined,
                    "required": undefined,
                    "type": "password",
                  }
                }
                size="large"
              >
                <FauxControl
                  className="css-2"
                  control={[Function]}
                  controlProps={
                    Object {
                      "aria-describedby": undefined,
                      "aria-invalid": undefined,
                      "aria-required": undefined,
                      "controlRef": undefined,
                      "disabled": undefined,
                      "readOnly": undefined,
                      "required": undefined,
                      "type": "password",
                    }
                  }
                  size="large"
                >
                  <FauxControl
                    className="css-2"
                    control={[Function]}
                  >
                    <div
                      className="css-4"
                    >
                      <glamorous(glamorous(input))
                        controlPropsIn={
                          Object {
                            "aria-describedby": undefined,
                            "aria-invalid": undefined,
                            "aria-required": undefined,
                            "controlRef": undefined,
                            "disabled": undefined,
                            "readOnly": undefined,
                            "required": undefined,
                            "type": "password",
                          }
                        }
                        key="control"
                        size="large"
                        type="password"
                      >
                        <input
                          className="css-15"
                          size="large"
                          type="password"
                        />
                      </glamorous(glamorous(input))>
                      <Underlay>
                        <div
                          className="css-29"
                        />
                      </Underlay>
                    </div>
                  </FauxControl>
                </FauxControl>
              </TextInput>
            </TextInput>
          </label>
        </div>
      </FormField>
    </FormField>
    <FormFieldDivider>
      <FormFieldDivider
        role="separator"
      >
        <div
          className="css-30"
          role="separator"
        />
      </FormFieldDivider>
    </FormFieldDivider>
    <FormField
      label="New Password"
      requiredText="Required"
    >
      <FormField>
        <div
          className="css-0"
        >
          <label>
            <glamorous(div)
              key="formField-2-textWrapper"
            >
              <div
                className="css-1"
              >
                <span
                  id="formField-2-labelText"
                >
                  New Password
                </span>
              </div>
            </glamorous(div)>
            <TextInput
              key="formField-2-control"
              type="password"
            >
              <TextInput
                control={[Function]}
                controlProps={
                  Object {
                    "aria-describedby": undefined,
                    "aria-invalid": undefined,
                    "aria-required": undefined,
                    "controlRef": undefined,
                    "disabled": undefined,
                    "readOnly": undefined,
                    "required": undefined,
                    "type": "password",
                  }
                }
                size="large"
              >
                <FauxControl
                  className="css-2"
                  control={[Function]}
                  controlProps={
                    Object {
                      "aria-describedby": undefined,
                      "aria-invalid": undefined,
                      "aria-required": undefined,
                      "controlRef": undefined,
                      "disabled": undefined,
                      "readOnly": undefined,
                      "required": undefined,
                      "type": "password",
                    }
                  }
                  size="large"
                >
                  <FauxControl
                    className="css-2"
                    control={[Function]}
                  >
                    <div
                      className="css-4"
                    >
                      <glamorous(glamorous(input))
                        controlPropsIn={
                          Object {
                            "aria-describedby": undefined,
                            "aria-invalid": undefined,
                            "aria-required": undefined,
                            "controlRef": undefined,
                            "disabled": undefined,
                            "readOnly": undefined,
                            "required": undefined,
                            "type": "password",
                          }
                        }
                        key="control"
                        size="large"
                        type="password"
                      >
                        <input
                          className="css-15"
                          size="large"
                          type="password"
                        />
                      </glamorous(glamorous(input))>
                      <Underlay>
                        <div
                          className="css-29"
                        />
                      </Underlay>
                    </div>
                  </FauxControl>
                </FauxControl>
              </TextInput>
            </TextInput>
          </label>
        </div>
      </FormField>
    </FormField>
    <FormField
      label="Repeat New Password"
      requiredText="Required"
    >
      <FormField>
        <div
          className="css-0"
        >
          <label>
            <glamorous(div)
              key="formField-3-textWrapper"
            >
              <div
                className="css-1"
              >
                <span
                  id="formField-3-labelText"
                >
                  Repeat New Password
                </span>
              </div>
            </glamorous(div)>
            <TextInput
              key="formField-3-control"
              type="password"
            >
              <TextInput
                control={[Function]}
                controlProps={
                  Object {
                    "aria-describedby": undefined,
                    "aria-invalid": undefined,
                    "aria-required": undefined,
                    "controlRef": undefined,
                    "disabled": undefined,
                    "readOnly": undefined,
                    "required": undefined,
                    "type": "password",
                  }
                }
                size="large"
              >
                <FauxControl
                  className="css-2"
                  control={[Function]}
                  controlProps={
                    Object {
                      "aria-describedby": undefined,
                      "aria-invalid": undefined,
                      "aria-required": undefined,
                      "controlRef": undefined,
                      "disabled": undefined,
                      "readOnly": undefined,
                      "required": undefined,
                      "type": "password",
                    }
                  }
                  size="large"
                >
                  <FauxControl
                    className="css-2"
                    control={[Function]}
                  >
                    <div
                      className="css-4"
                    >
                      <glamorous(glamorous(input))
                        controlPropsIn={
                          Object {
                            "aria-describedby": undefined,
                            "aria-invalid": undefined,
                            "aria-required": undefined,
                            "controlRef": undefined,
                            "disabled": undefined,
                            "readOnly": undefined,
                            "required": undefined,
                            "type": "password",
                          }
                        }
                        key="control"
                        size="large"
                        type="password"
                      >
                        <input
                          className="css-15"
                          size="large"
                          type="password"
                        />
                      </glamorous(glamorous(input))>
                      <Underlay>
                        <div
                          className="css-29"
                        />
                      </Underlay>
                    </div>
                  </FauxControl>
                </FauxControl>
              </TextInput>
            </TextInput>
          </label>
        </div>
      </FormField>
    </FormField>
  </div>
</glamorous(div)>
`;
